src/main/kotlin/kotlinx/kover/KoverPlugin.kt

```diff
@@ -148,6 +150,24 @@ class KoverPlugin : Plugin<Project> {
         htmlReportTask.binaryReportFiles.set(binariesProvider)
         verificationTask.binaryReportFiles.set(binariesProvider)
 
+        val smapProvider = provider {
+            val files = tasks.withType(Test::class.java)
+                .map { t -> t.extensions.getByType(KoverTaskExtension::class.java) }
+                .filter { e -> e.isEnabled }
+                .map { e -> e.smapFile.orNull }
+                /*
+                 Binary reports and SMAP files have same ordering for IntelliJ engine:
+                    * SMAP file is null if coverage engine is a JaCoCo by default - in this case property is unused
+                    * SMAP file not creates by JaCoCo - property is unused
+                    * test task have no sources - in this case binary report and SMAP file not exists
+                 */
+                .filter { f -> f?.exists() == true }
```

Comment:
super nit -- when working with nulls, `f?.exists() ?: true` can be more readab;e